### PR TITLE
[Async] refactored async-api, to be more developerfriendly

### PR DESCRIPTION
**Pull request by plusgut**

### CODE REVIEW

File: src/components/Async.tsx [hunk 1]
```diff
@@ -1,79 +1,87 @@
 import type { Props } from "../index";
 import type ComponentInstance from "../instances/types/Component/Instance";
 import type { ApplicationElement } from "../interfaces/component";
-import type { invokeGuard } from "../interfaces/renderOptions";
+// import type { invokeGuard } from "../interfaces/renderOptions";
```
Comment (gaggala) [inline src/components/Async.tsx:4]: should this comment be deleted?

Comment (plusgut) [reply src/components/Async.tsx:4]: yep, thanks for the catch

File: test/integration/render/async.spec.tsx [hunk 2]
```diff
@@ -264,4 +297,55 @@ describe("<Animate />", () => {
...
+    expect(container.childNodes.length).toBe(1);
+    expect((container.childNodes[0] as HTMLElement).tagName).toBe("SPAN");
+
+    await tick(1);
+
+    expect((container.childNodes[0] as HTMLElement).tagName).toBe("DIV");
+    expect((container.childNodes[0] as HTMLElement).innerHTML).toBe("foo");
+  });
+
+  it("When promise gets rejected, the exception will bubble to the try component", async () => {
```
Comment (gaggala) [inline test/integration/render/async.spec.tsx:326]: correct description?

Comment (plusgut) [reply test/integration/render/async.spec.tsx]: oops, that slipped through me. thanks

**Pull request merged**